dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -603,141 +600,113 @@ lateral jsonb_each(
       String teaUid = tea.getUid();
       String teaValueCol = quote(teaUid);
-      String teaCol = quote(teaUid + "ATT");
 
-      sql.append(" inner join trackedentityattributevalue ")
-          .append(teaValueCol)
-          .append(" on ")
-          .append(teaValueCol + ".trackedentityid")
-          .append(" = TE.trackedentityid ")
-          .append(" inner join trackedentityattribute ")
-          .append(teaCol)
-          .append(" on ")
-          .append(teaValueCol + ".trackedentityattributeid")
-          .append(EQUALS)
-          .append(teaCol + ".trackedentityattributeid")
+      fromBuilder
+          .append(hlp.whereAnd())
+          .append(" TE.trackedentityid is not null ") // filters out results from event programs
           .append(AND)
-          .append(teaCol + ".UID")
-          .append(EQUALS)
-          .append(singleQuote(teaUid));
-
-      sql.append(
-          getAttributeFilterQuery(
-              mapSqlParameterSource,
-              queryItem.getValue(),
-              teaCol,
-              teaValueCol,
-              tea.getValueType().isNumeric()));
+          .append(SPACE)
+          .append(
+              getAttributeFilterQuery(
+                  mapSqlParameterSource,
+                  queryItem.getValue(),
+                  teaValueCol,
+                  tea.getValueType().isNumeric()));
     }
-
-    return sql.toString();
+    return fromBuilder.toString();
   }
 
   private String getAttributeFilterQuery(
       MapSqlParameterSource mapSqlParameterSource,
       List<QueryFilter> filters,
-      String teaCol,
       String teaValueCol,
       boolean isNumericTea) {
 
     if (filters.isEmpty()) {
       return "";
     }
 
-    StringBuilder query = new StringBuilder(AND);
-    // In SQL the order of expressions linked by AND is not
-    // guaranteed.
-    // So when casting to number we need to be sure that the value
-    // to cast is really a number.
-    if (isNumericTea) {
-      query
-          .append(" case when ")
-          .append(lower(teaCol + ".valueType"))
-          .append(" in (")
-          .append(
-              NUMERIC_TYPES.stream()
-                  .map(Enum::name)
-                  .map(StringUtils::lowerCase)
-                  .map(SqlUtils::singleQuote)
-                  .collect(Collectors.joining(",")))
-          .append(")")
-          .append(" then ");
-    }
-
+    StringBuilder query = new StringBuilder();
     List<String> filterStrings = new ArrayList<>();
 
     for (int i = 0; i < filters.size(); i++) {
       QueryFilter filter = filters.get(i);
-      StringBuilder filterString = new StringBuilder();
       final String queryCol =
           isNumericTea ? castToNumber(teaValueCol + ".value") : lower(teaValueCol + ".value");
       int itemType = isNumericTea ? Types.NUMERIC : Types.VARCHAR;
       String parameterKey = "attributeFilter_%s_%d".formatted(teaValueCol.replace("\"", ""), i);
```

Comment:
This looks odd. We pass the quoted UID `String teaValueCol = quote(teaUid);` in to then unquote it. We should be able to pass in the UID and quote it only when/where necessary, right?